CMakeLists.txt

```diff
@@ -106,6 +108,7 @@ ExternalProject_Add(${ZLIB_TARGET}
        BINARY_DIR ${BUILD_DIR}/zlib_build
        INSTALL_DIR ${CMAKE_INSTALL_PREFIX}
        CONFIGURE_COMMAND ${CMAKE_COMMAND} ${CMAKE_CORE_BUILD_FLAGS} -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR> <SOURCE_DIR>
+       BUILD_COMMAND $(MAKE)
```

Comment:
```
BUILD_COMMAND:
Overrides the default build command (generator expressions are supported). If this option is not given, the default build command will be chosen to integrate with the main build in the most appropriate way (e.g. using recursive make for Makefile generators or cmake --build if the project uses a CMake build). This option can be specified with an empty string as the command to make the build step do nothing.
```
So I do not understand why it would change the parallelization and I'm not sure where MAKE is defined.
If it's empty it skips the build.

Reply:
I think by default, it's treating the child projects as CMAKE projects (which they are), but this results in them being built single threaded as they are built with `cmake --build` rather than make.

Empirically, $MAKE is always set if the build is started with `make`. I've just confirmed that it also works correctly when invoked with `cmake --build` too.

On investigation, it feels like building with `cmake- -build` is achieving a bit more parallelism, so I'll use that in the Docker build scripts instead of make.